code/FulpstationCode/sec_borg_files/fulp_sec_borg.dm

```diff
@@ -0,0 +1,477 @@
+//Pepperspray Module
+
+/obj/item/reagent_containers/spray/pepper/cyborg
+	name = "Integrated Pepperspray"
+	desc = "An integrated pepperspray synthesizer. Use for blinding criminal scum. Utilizes your power supply to synthesize capsaicin spray over time."
+	reagent_flags = NONE
+	volume = 50
+	list_reagents = list(/datum/reagent/consumable/condensedcapsaicin = 50)
+	var/charge_cost = 50
+	var/generate_amount = 5
+	var/generate_type = /datum/reagent/consumable/condensedcapsaicin
+	var/last_generate = 0
+	var/generate_delay = 50	//deciseconds
+	var/upgraded = FALSE
+	can_fill_from_container = FALSE
+
+// Fix pepperspraying yourself
+/obj/item/reagent_containers/spray/pepper/cyborg/afterattack(atom/A as mob|obj, mob/user)
+	if (A.loc == user)
+		return
+	. = ..()
+
+/obj/item/reagent_containers/spray/pepper/cyborg/Initialize()
+	. = ..()
+	START_PROCESSING(SSfastprocess, src)
+
+/obj/item/reagent_containers/spray/pepper/cyborg/Destroy()
+	STOP_PROCESSING(SSfastprocess, src)
+	return ..()
+
+/obj/item/reagent_containers/spray/pepper/cyborg/process()

```

Comment:
I'd personally just make secborgs refill their pepper spray with wall mounted capsaicin dispensers instead of having it regen on its own, just as other borgs would refill their welders with welding tanks and refill their extinguishers with water tanks.

Reply:
No, this is fine. 5 seconds to regain 5U at the cost of energy is acceptable.